src/app_engine/gcm_notify.py

```diff
@@ -44,94 +45,119 @@ def get_gcm_api_key():
   return GCM_API_KEY
 
 
-def send_gcm_messages(gcm_ids, message, collapse_key):
-  """Sends the message to each endpoint specified in gcm_ids.
+class GCMMessage(object):
+  def __init__(self, gcm_id, registration_id):
+    self.gcm_id = gcm_id
+    self.registration_id = registration_id
 
-  Makes a synchronous post to the GCM server with the appropriate payload
-  format.
+  def get_gcm_payload(self):
+    payload_dict = {
+        'data': self.get_data(),
+        'registration_ids': [self.gcm_id],
+        'time_to_live': GCM_TIME_TO_LIVE_IN_SECONDS
+    }
+    collapse_key = self.get_collapse_key()
+    if collapse_key:
+      payload_dict['collapse_key'] = collapse_key
+    return json.dumps(payload_dict)
+
+  def get_data(self):
+    return {GCM_MESSAGE_REGISTRATION_ID_KEY: self.registration_id}
+
+  def get_collapse_key(self):
+    return None
+
+
+class GCMInviteMessage(GCMMessage):
+  def __init__(self, gcm_id, registration_id, room_id, caller_id, metadata):
+    super(GCMInviteMessage, self).__init__(gcm_id, registration_id)
+    self.room_id = room_id
+    self.caller_id = caller_id
+    self.metadata = metadata
+
+  def get_data(self):
+    data = super(GCMInviteMessage, self).get_data()
+    data[GCM_MESSAGE_TYPE_KEY] = GCM_MESSAGE_TYPE_INVITE
+    data[constants.PARAM_ROOM_ID] = self.room_id,
+    data[constants.PARAM_CALLER_ID] = self.caller_id,
+    data[constants.PARAM_METADATA] = self.metadata,
+    return data
+
+  def get_collapse_key(self):
+    return self.room_id
+
+
+class GCMByeMessage(GCMMessage):
+  def __init__(self, gcm_id, registration_id, room_id, reason):
+    super(GCMByeMessage, self).__init__(gcm_id, registration_id)
+    self.gcm_id = gcm_id
+    self.registration_id = registration_id
```

Comment:
Not the same. Registration ID is assigned by AppRTC server per registration. GCM ID is returned by GCM library.